### PR TITLE
Experience Composer API enhancements

### PR DESCRIPTION
**Pull request by manchuck**
#### What is this PR doing?

Adding the render

#### How should this be manually tested?

```js 
opentok.listRenders({}, (err, renders) => {
    if (err) {
        console.log("There was an error")
        console.log(err);
        return;
    }

    console.log(renders);
})

opentok.getRender('<render id>', (err, render) => {
    if (err) {
        console.log("There was an error")
        console.log(err);
        return;
    }

    console.log(render);
})

opentok.startRender(
    {
        sessionId: sessionId,
        token: token,
        url: '<callback url>',
    }, 
    (err, render) => {
        if (err) {
            console.log("There was an error")
            console.log(err);
            return;
        }

        console.log(render);
    }
)

opentok.stopRender('<render id>', (err, render) => {
    if (err) {
        console.log("There was an error")
        console.log(err);
        return;
    }

    console.log(render);
})
```

#### What are the relevant tickets?

[DEVX-6495](https://jira.vonage.com/browse/DEVX-6495)

### CODE REVIEW

File: lib/opentok.js [hunk 1]
```diff
@@ -477,6 +478,168 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+   *   <li><code>render</code> &mdash; The {@link Render} object.</li>
+   * </ul>
+   *
+   * @method #getArchive
+   * @memberof OpenTok
+   */
+  this.getRender = render.getRender.bind(null, apiConfig);
+
+  /**
+  * Starts rendering an OpenTok session.
```
Comment (jeffswartz) [inline lib/opentok.js:551]: Please copy (with any required edits) the docs for the REST method: https://tokbox.com/developer/rest/#starting_experience_composer (for this description and the docs for the options). For example:

```suggestion
  * Starts an Experience Composer for an OpenTok session. The Experience Composer
  * instance is represented as a Render object.
  * For more information, see the
  * <a href="https://tokbox.com/developer/guides/experience-composer/">Experience Composer developer guide</a>.
```

File: lib/render.js [hunk 2]
```diff
@@ -0,0 +1,221 @@
+/* eslint-env es2018 */
+const errors = require('./errors');
+const request = require('request');
+const { version } = require('../package.json');
+const _ = require('lodash');
+const generateJwt = require('./generateJwt.js');
+
+/**
+* An object representing an OpenTok render.
```
Comment (jeffswartz) [inline lib/render.js:9]:
```suggestion
* An object representing an Experience Composer renderer.
```

File: lib/render.js [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+* @property {String} projectId
+*   The ID of the project for the render.
+* 
+* @property {String} sessionId
+*   The ID of the session being rendered into.
+* 
+* @property {Number} createdAt
+*   The time at which the render was created, in milliseconds since the UNIX epoch.
+** 
+* @property {Number} upddatedAt
```
Comment (jeffswartz) [inline lib/render.js:23]:
```suggestion
* @property {Number} updatedAt
```

File: lib/render.js [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+*   The time at which the render was created, in milliseconds since the UNIX epoch.
+*
+* @property {String} url
+*   A publically reachable URL controlled by the customer and capable of generating 
+*   the content to be rendered without user intervention.
+* 
+* @proerpty {String} status
+*   Current status of for the render. Will be one of `started`, `stopped` or `failed`
```
Comment (jeffswartz) [inline lib/render.js:31]: The REST docs also mention "starting". Should these be strings (e.g., `"started"`)?

File: lib/render.js [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+    if (exception instanceof SyntaxError) {
+      callback(new errors.RequestError('Cannot decode JSON response'));
+      return;
+    }
+
+    callback(new Error('Uknown Error'));
+  }
+};
+
+exports.listRenders = (config, options, callback) => {
```
Comment (jeffswartz) [inline lib/render.js:149]: Please add docs.

File: lib/render.js [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    {
+      path: 'v2/project/' + config.apiKey + '/render',
+      config: config,
+      params: params
+    },
+    _.partial(handleResponse, cb)
+  );
+};
+
+exports.getRender = (config, renderId, callback) => {
```
Comment (jeffswartz) [inline lib/render.js:177]: Please add docs.

File: lib/render.js [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+  api(
+    {
+      path: 'v2/project/' + config.apiKey + '/render/' + renderId,
+      config: config
+    },
+    _.partial(handleResponse, cb)
+  );
+};
+
+exports.startRender = (config, options, callback) => {
```
Comment (jeffswartz) [inline lib/render.js:189]: Please add docs.

File: lib/render.js [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+        maxDuration: _.get(opts, 'maxDuration', 1800),
+        resolution: _.get(opts, 'resolution', '1280x720'),
+        statusCallbackUrl: _.get(opts, 'statusCallbackUrl')
+      }
+    },
+    _.partial(handleResponse, cb)
+  );
+};
+
+exports.stopRender = (config, renderId, callback) => {
```
Comment (jeffswartz) [inline lib/render.js:210]: Please add docs.

File: lib/opentok.js [hunk 9]
```diff
@@ -477,6 +478,168 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+  *   </li>
+  *   
+  *   <li>
+  *     <code>resolution</code> (String) &mdash; Resolution of the display area for the composition.
+  *   </li>
+  *   
+  *   <li>
+  *     <code>statusCallbackUrl</code> (String) &mdash; URL of the customer service where the 
+  *     callbacks will be received.
+  *   </li>
```
Comment (jeffswartz) [inline lib/opentok.js:592]: We are not documenting this REST API option. Developers should use account portal to register callbacks. See https://tokbox.com/developer/guides/experience-composer/#configuring-callbacks.

**Review by jeffswartz — CHANGES_REQUESTED** [review]

**Review by hamzanasir — APPROVED** [review]
Other than adding the docs that @jeffswartz commented it looks good to me!

**Review by jeffswartz — APPROVED** [review]

File: lib/opentok.js [hunk 10]
```diff
@@ -477,6 +478,166 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+   *     <code>sessionId</code> &mdash; Specify the ID of a session in order to retrieve renders
+   *     specifically for that session. This property is optional. When no session ID is specified,
+   *     then the method will return renders from any session created with your API key.
+   *   </li>
+   *
+   * </ul>
+   *
+   * <p>If you don't pass in an <code>options</code> argument,
+   * the method returns up to 1000 renders
+   * starting with the first render recorded.
```
Comment (jeffswartz) [inline lib/opentok.js:511]:
```suggestion
   * the method returns up to 1000 renders, starting with the first render recorded.
```

File: lib/opentok.js [hunk 11]
```diff
@@ -477,6 +478,166 @@ OpenTok = function (apiKey, apiSecret, env) {
...
+  *
+  * </ul>
+  *
+  * @method #startRender
+  * @memberof OpenTok
+  */
+  this.startRender = render.startRender.bind(null, apiConfig);
+
+  /**
+   * Stops an OpenTok render that is being rendered.
```
Comment (jeffswartz) [inline lib/opentok.js:616]:
```suggestion
   * Stops an experience composer that is being rendered.
```

**Review by jeffswartz — APPROVED** [review]

**Pull request merged**